dai.md

```diff
@@ -42,31 +45,119 @@ module DAI
 
     syntax DaiStep ::= ExceptionStep
  // --------------------------------
+    rule <k> Dai . _:DaiStep => Dai . exception ... </k> [owise]
 
     syntax DaiStep ::= "transfer" Address Wad
  // -----------------------------------------
+    rule <k> Dai . transfer ACCOUNT_SRC AMOUNT => . ... </k>
+         <msg-sender> ACCOUNT_SRC </msg-sender>
+         <dai-balance>
+           ...
+           ACCOUNT_SRC |-> BALANCE_SRC
+           ...
+         </dai-balance>
+      requires BALANCE_SRC >=Int AMOUNT
+
+    rule <k> Dai . transfer ACCOUNT_DST AMOUNT => . ... </k>
+         <msg-sender> ACCOUNT_SRC </msg-sender>
+         <dai-balance>
+           ...
+           ACCOUNT_SRC |-> (BALANCE_SRC => BALANCE_SRC -Int AMOUNT)
+           ACCOUNT_DST |-> (BALANCE_SRC => BALANCE_SRC +Int AMOUNT)
+           ...
+         </dai-balance>
+      requires ACCOUNT_SRC =/=K ACCOUNT_DST
+       andBool BALANCE_SRC >=Int AMOUNT
 
     syntax DaiStep ::= "transferFrom" Address Address Wad
  // -----------------------------------------------------
+    rule <k> Dai . transferFrom ACCOUNT_SRC ACCOUNT_SRC AMOUNT => . ... </k>
+         <dai-balance>
+           ...
+           ACCOUNT_SRC |-> BALANCE_SRC
+           ...
+         </dai-balance>
+      requires BALANCE_SRC >=Int AMOUNT
+
+    rule <k> Dai . transferFrom ACCOUNT_SRC ACCOUNT_DST AMOUNT => . ... </k>
+         <dai-balance>
+          ...
+          ACCOUNT_SRC |-> (BALANCE_SRC => BALANCE_SRC -Int AMOUNT)
+          ACCOUNT_DST |-> (BALANCE_DST => BALANCE_DST +Int AMOUNT)
+          ...
+        </dai-balance>
+        <dai-allowance>
+          ...
+          { ACCOUNT_SRC -> ACCOUNT_DST } |-> (ALLOWANCE_SRC_DST => ALLOWANCE_SRC_DST -Int AMOUNT)
```

Comment:
I don't recognize this syntax, `{ _ -> _ }`, where does this come from?

Reply:
It's defined further up in the file.